x-pack/plugins/timelines/public/components/actions/timeline/cases/add_to_case_action_button.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import {
+  EuiPopover,
+  EuiButtonIcon,
+  EuiContextMenuPanel,
+  EuiText,
+  EuiContextMenuItem,
+  EuiToolTip,
+} from '@elastic/eui';
+import { AddToCaseActionProps } from './add_to_case_action';
+import { useAddToCase } from '../../../../hooks/use_add_to_case';
+import { ActionIconItem } from '../../action_icon_item';
+import * as i18n from './translations';
+
+const AddToCaseActionButtonComponent: React.FC<AddToCaseActionProps> = ({
+  ariaLabel = i18n.ACTION_ADD_TO_CASE_ARIA_LABEL,
+  ecsRowData,
+  useInsertTimeline,
+  casePermissions,
+  appId,
+  onClose,
+}) => {
+  const {
+    addNewCaseClick,
+    addExistingCaseClick,
+    isDisabled,
+    userCanCrud,
+    isEventSupported,
+    openPopover,
+    closePopover,
+    isPopoverOpen,
+  } = useAddToCase({ ecsRowData, useInsertTimeline, casePermissions, appId, onClose });
+  const tooltipContext = userCanCrud
+    ? isEventSupported
+      ? i18n.ACTION_ADD_TO_CASE_TOOLTIP
+      : i18n.UNSUPPORTED_EVENTS_MSG
+    : i18n.PERMISSIONS_MSG;
+  const items = useMemo(
```

Comment:
if we could do this in the object format instead of jsx, we can use this as a context menu component. i have a commit thats gonna mess your day up, but i need it for this work: https://github.com/elastic/kibana/issues/107917 

> Use EuiContextMenuPanel for simple, non-nested context menus. The below pagination example has no nesting and no title.

so the issue with having the jsx type for the menu items (`EuiContextMenuPanel` vs using `EuiContextMenu` with the `panels` prop) is you cant nest it. the updates to the alert table call for nesting the action in the `...` `alert_context_menu`.

how it is (pretend text is formatted):
<img width="324" alt="Screen Shot 2021-08-11 at 1 46 55 PM" src="https://user-images.githubusercontent.com/6935300/129201028-62c3547a-4789-4878-9e91-09dea669fb93.png">

how it needs to be when we use the props instead of the jsx:
![gif](https://user-images.githubusercontent.com/6935300/129201063-b31103ff-471b-4337-9812-cea921cd1ed4.gif)